### PR TITLE
Fix deprecations

### PR DESCRIPTION
**Pull request by GuilhemN**
Fixes https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1897

### CODE REVIEW

**Comment by GuilhemN** [comment]
Remaining deprecations are unrelated.

File: Routing/Loader/RestRouteLoader.php [hunk 1]
```diff
@@ -117,6 +118,12 @@ private function getControllerLocator($controller)
         if ($this->container->has($controller)) {
             // service_id
             $prefix = $controller.':';
+
+             // sf > 4.1 support
+            if (method_exists(Kernel::class, 'getAnnotatedClassesToCompile')) {
```
Comment (xabbuh) [inline Routing/Loader/RestRouteLoader.php]: In the `Kernel` class we could more easily check the values of the version related constants. :)

**Comment by JustBlackBird** [comment]
@GuilhemN Could you please release a new minor version of the bundle with the fix?

File: Routing/Loader/RestRouteLoader.php (hunk 1 above)
Comment (GuilhemN) [reply Routing/Loader/RestRouteLoader.php]: Indeed, see https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1907 :)

**Pull request merged**